File: rest/message/list-get-example-2/list-get-example-2.5.x.rb
# Get twilio-ruby from twilio.com/docs/ruby/install
require 'twilio-ruby'

# Get your Account SID and Auth Token from twilio.com/console
account_sid = 'ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
auth_token = 'your_auth_token'

# Initialize Twilio Client
@client = Twilio::REST::Client.new(account_sid, auth_token)

@client.messages.list(
  to: 'to_number',
  from: 'from_number',
  date_sent: Time.new('2015-04-01T00:00:00Z')
).each do |message|
  puts message.body
end

File: rest/message/list-get-example-2/list-get-example-2.5.x.php
<?php
// Get the PHP helper library from https://twilio.com/docs/libraries/php
require_once '/path/to/vendor/autoload.php'; // Loads the library
use Twilio\Rest\Client;

// Your Account Sid and Auth Token from twilio.com/user/account
// To set up environmental variables, see http://twil.io/secure
$sid = getenv('TWILIO_ACCOUNT_SID');
$token = getenv('TWILIO_AUTH_TOKEN');

$client = new Client($sid, $token);

// Loop over the list of messages and echo a property for each one
$params = array('to' => 'to_number', 'from' => 'from_number', 'dateSent' => date('2016-09-06'));
$result = $client->messages->read($params);

foreach ($result as $message) {
    echo $message->body;
}
